src/audio/coreaudio/SDL_coreaudio.m

```diff
@@ -1151,6 +1151,138 @@ output device (in which case we'll try again). */
     return (this->hidden->thread != NULL) ? 0 : -1;
 }
 
+#if !MACOSX_COREAUDIO
+static int
+COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
+{
+    AVAudioSession* session = [AVAudioSession sharedInstance];
+    spec->freq = [session sampleRate];
```

Comment:
Both name and spec aren't initialized, so at minimum:

```
if (name != NULL) {
    *name = NULL; /* Or an allocated string */
}
SDL_zerop(spec);
```

Reply:
The `name` part applies to the macosx case, too.

Reply:
I think the if statement around line 1242 already pretty much does that. Should I change it to `if (name != NULL)` rather than just `if (name)` though?